src/Transport/Sending/IBatcher.cs

```diff
@@ -1,10 +1,11 @@
 namespace NServiceBus.Transport.AzureServiceBus
 {
     using System.Collections.Generic;
+    using System.Threading.Tasks;
     using Transport;
 
     interface IBatcherInternal
     {
-        IList<BatchInternal> ToBatches(TransportOperations operations);
+        Task<List<BatchInternal>> ToBatches(TransportOperations operations);
```

Comment:
The original implementation (`hotfix-7.2.4` did exactly that while using `Lazy` instead of `AsyncLazy`).
Downside was a single blocking invocation. I can revert changes to the API and prevent Task propagation so that `IBatcher` and `ITopologySectionManagerInternal` do not bleed `Task` because of the `BuildTopicBundles` method.